### DISCUSSION TITLE
Storing into Application Local State in Algorand

### DISCUSSION THREAD

**Discussion by No-Cash-7970** [reach-sh/reach-lang#1052] [Q&A] [answered]
With Algorand, how do I store a value into the opted-in application local state with a key? It seems Reach's `Map` does not support keys for Algorand. I'm looking for a way to store data into the application local state similar to TEAL's `app_global_put`, which takes a key and a value.

**Comment by jeapostrophe** [comment, marked as answer]
The answer is `Map`. You decide the data structure(s) you want to store with users and Reach will figure out how to make the data-structure fit with the restrictions imposed by Algorand.

**Reply by No-Cash-7970** [reply]
How do you create the data structure? Do you use another `Map`? Something like `Map(Address, Map(Bytes(8), UInt))`?

**Reply by jeapostrophe** [reply]
You can use any type you want, but I think you'll want an object. https://docs.reach.sh/rsh/compute/#ref-programs-types
